home/gettingstarted/0.7/javausage.md

```diff
@@ -12,66 +12,117 @@ outof: 50
 
     import java.util.HashMap;
     import java.util.Iterator;
     import java.util.LinkedList;
 
-We extend the `OfflineReport` class -- you can read more about it in the [configuration section](/home/gettingstarted/0.7/configuration/index.html):
-
-    public class MyHistogramTest extends OfflineReport {
+We extend the `JBench.OfflineReport` class -- you can read more about it in the [configuration section](/home/gettingstarted/0.7/configuration/index.html):
 
-Next, we implement the `javaPersistor` method -- you can read more about this in the [configuration section](/home/gettingstarted/0.7/configuration/index.html):
-
-      public Persistor javaPersistor() {
-        return new SerializationPersistor();
-      }
+    public class MyHistogramTest extends JBench.OfflineReport {
 
 Benchmarks in a test are organized into groups.
-In the Scala frontend, we encoded the groups using nested closures.
-In the Java frontend, we encode them by implementing the `Group` interface.
-We now declare a `List` benchmark group:
-
-      public class List implements Group {
-
-Next, we declare a `Using` class for the benchmark.
-In the Scala frontend, this is achieved with the `using` block, as described in the [simple microbenchmark example](/home/gettingstarted/0.7/simplemicrobenchmark/index.html).
-
-        public class groupBy
-        implements Using<LinkedList<Integer>, HashMap<Integer, LinkedList<Integer>>> {
-
-To override the default configuration for a `Using` class or a `Group`,
-we declare a final `config` field of the type `JContext` (more about custom per-group configurations [here](/home/gettingstarted/0.7/configuration/index.html)):
-
-          public final JContext config = JContext.create()
-            .put("exec.benchRuns", 20)
-            .put("exec.independentSamples", 1)
-            .put("exec.outliers.covMultiplier", 1.5)
-            .put("exec.outliers.suspectPercent", 40);
-
-Next, we declare a generator for the `groupBy` class (see more about generators in the [generator section](/home/gettingstarted/0.7/generators/index.html)):
-
-          public JavaGenerator<LinkedList<Integer>> generator() {
-            JavaGenerator<Integer> sizes = new SingleGen("size", 5000000);
-            return new CollectionGenerators(sizes).lists();
+In the Scala frontend, we encode the groups using nested closures.
+In the Java frontend, we encode them by placing the `benchmark` annotation 
+that argument denotes the group.
+We now declare benchmark snippet stub in a `list` group:
+
+      @benchmark("list")
+      public void histogram() {
+      
+      }
+      
+Next, we want a custom context:
+
+      public final Context config = new ContextBuilder()
+          .put("exec.benchRuns", 20)
+          .put("exec.independentSamples", 1)
+          .put("exec.outliers.covMultiplier", 1.5)
+          .put("exec.outliers.suspectPercent", 40)
+          .build();
+          
+We want it not only for our histogram method 
```

Comment:
Let's assume that we want to apply this custom `config` not only to our `histogram` method,
    but to all benchmark snippets in the `list` group.
    We do this by adding the optional `scopes` annotation to the `MyHistogramTest` class, as follows: